### ISSUE TITLE
QoL: Allow `Label` to have a box surrounding it

### ISSUE THREAD

**Issue by Datseris** [MakieOrg/Makie.jl#2837]
Would be nice if it is possible that `Label` can also have a box that surrounds its text and is scaled appropriately. I.e., so that it looks like a legend box, but without any legend entries, just text.

Currently there is no straightforward way to do this.

**Comment by asinghvi17** [comment]
We have a bbox for the label already - ideally there would be some padding involved if we were to plot a background to text.  

That padding might want to depend on the size of the text (i.e., margin = 10% the size of the smallest dimension of the text, OR 1 lineheight, ...).  What are some other possible inputs you could think of?  Could we use the `Auto`/`Relative` structs here to further indicate padding width?

**Comment by Datseris** [comment]
I thought `bbox` refers to the bounding box, not making an actually drawn box. I was wrong?

**Comment by asinghvi17** [comment]
Yes that's what I meant, sorry - it would be trivial to plot the background but it would conform tightly to the character, which would look kinda bad.

**Comment by Datseris** [comment]
ah, i see. then yes I agree with your suggestion of padding by a user-provided keyword, by default 110 points (i thikn percentage padding isn't a good idea here, because i would expect same distance between characters and label box regadlesss of if I have 10 or 120 characters)

**Comment by ffreyer** [comment]
@jkrumbiegel Isn't there a way to do this with `Box`? If that's simple enough maybe we can document that instead?
